src/sparseml/pytorch/sparsification/distillation/modifier_per_layer.py

```diff
@@ -354,7 +356,7 @@ def update(
         steps_per_epoch: int,
     ):
         super().update(module, optimizer, epoch, steps_per_epoch)
-        if epoch >= self.end_epoch:
+        if self.end_epoch > 0 and epoch >= self.end_epoch:
```

Comment:
nit: maybe this reads better?
```suggestion
        if 0 < self.end_epoch <= epoch:
```

Reply:
Yep I do like that better